PhotoBooth/CaptureSession.swift

```diff
@@ -1,19 +1,143 @@
 import UIKit
+import AVFoundation
+
+protocol PhotoCaptureable {
+
+    func captureImage() -> UIImage
 
-protocol ImageCapturing {
-    
-    func getImage() -> UIImage
-    
     func configurePreview(view: UIView)
+
+    func switchCamera()
 }
 
-class CaptureSession: ImageCapturing {
-    
-    func getImage() -> UIImage {
+class CaptureSession: NSObject, PhotoCaptureable, AVCapturePhotoCaptureDelegate {
+
+    private var avSession = AVCaptureSession()
+
+    private var backCamera: AVCaptureDevice?
+    private var frontCamera: AVCaptureDevice?
+    private var currentCamera: AVCaptureDevice?
+
+    private var photoOutput: AVCapturePhotoOutput?
+    private let photoSessionPreset = AVCaptureSession.Preset.photo
+
+    private var cameraPreviewLayer: AVCaptureVideoPreviewLayer?
+
+    private var images = [UIImage]()
+
+    private func setupCaptureSession() {
+        setupPhotoCaptureSession()
+        setupDevices()
+        setUpCaptureSessionInput(position: .front)
+    }
+
+    func captureImage() -> UIImage {
         return UIImage()
     }
-    
+
     func configurePreview(view: UIView) {
-        
+        setupCaptureSession()
+        setupPreviewLayer(view: view)
+        startRunningCaptureSession()
+    }
+
+    // This function sets up a switch to change the camera in use depending on current position when called.
+    private func setUpCaptureSessionInput(position: AVCaptureDevice.Position) {
+        switch position {
+        case .back:
+            currentCamera = backCamera
+        case .front:
+            currentCamera = frontCamera
+        default:
+            currentCamera = backCamera
+        }
+
+        setupInputOutput()
+    }
+
+    // Mark:- @objc functions for buttons
+    // This is the function that will be called when the take photo button is pressed.
+    @objc func takePhoto(){
+        avSession.beginConfiguration()
+        avSession.sessionPreset = photoSessionPreset
+        avSession.commitConfiguration()
```

Comment:
does this config have to happen every time?

Reply:
so I was looking at the documentation and it looks like any settings like flash or camera orientation or position or lighting or focus etc... needs to be committed as the configuration before you take a photo or video.